### DISCUSSION TITLE
Improve Small File Performance

### DISCUSSION THREAD

**Discussion by zaxlofful** [abraunegg/onedrive#2489] [Ideas]
I was wondering what could be done to help improve write lots of small files to the OS, I am not sure if it can be batched or use multiple threads for multiple files; but it seems very slow comparative to other sync applications.

What are the options and what does this application already do under the hood?

**Comment by abraunegg** [comment]
@ZaxLofful 
> I was wondering what could be done to help improve write lots of small files to the OS, I am not sure if it can be batched or use multiple threads for multiple files; but it seems very slow comparative to other sync applications.

Please can you read:
* https://github.com/abraunegg/onedrive/issues/232
* https://github.com/abraunegg/onedrive/discussions/2478

**Reply by zaxlofful** [reply]
Thanks!
